### PR TITLE
wifi: esp32: allow wifi symbols into flash

### PR DESCRIPTION
**Pull request by sylvioalves**
When BT and WiFi coexists, IRAM usage increases a lot.
Add configuration that allow wifi symbols
to be placed in flash, freeing space in IRAM.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: drivers/wifi/esp32/Kconfig.esp32 [hunk 1]
```diff
@@ -225,4 +225,25 @@ config ESP32_WIFI_SW_COEXIST_ENABLE
...
+	default y
+	help
+	  Select this option to place frequently called Wi-Fi library RX functions in IRAM.
+	  When this option is disabled, more than 17Kbytes of IRAM memory will be saved
+	  but Wi-Fi performance will be reduced.
+
+config ESP32_WIFI_SW_COEXIST_ENABLE
+	bool
+	help
+	  Software controls WiFi/Bluetooth coexistence. Not supported yet.
```
Comment (dcpleung) [inline drivers/wifi/esp32/Kconfig.esp32:247]: This is not related to the PR. Please remove.

**Review by dcpleung — CHANGES_REQUESTED** [review]

File: drivers/wifi/esp32/Kconfig.esp32 (hunk 1 above)
Comment (sylvioalves) [reply drivers/wifi/esp32/Kconfig.esp32]: Yes, fixed, thanks.

**Review by dcpleung — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by mniestroj — APPROVED** [review]
